From 7768322dd63667101029106562d253346472f944 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Sun, 20 Jun 2021 15:28:35 -0700
Subject: [PATCH] Bump Lotus version to 1.0.0 (#201)

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 5523955d97961f98c4656d26bbf69ed954248d75
Recorded-replay-commit: a6554cbcad6d69fc8e44f179c568cfda68dc8eb6

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33a0bed6a3..609db857c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,13 +1,13 @@
 # Copyright (c) 2017 The Bitcoin developers
 
-cmake_minimum_required(VERSION 3.25)
+cmake_minimum_required(VERSION 3.18)
 
 set(CMAKE_USER_MAKE_RULES_OVERRIDE
 	"${CMAKE_SOURCE_DIR}/cmake/modules/OverrideInitFlags.cmake"
 )
 
 project(lotus
-	VERSION 0.34.1
+	VERSION 1.0.0
 	DESCRIPTION "Lotus is a full node implementation of the Lotus protocol."
 	HOMEPAGE_URL "https://www.givelotus.org"
 )
@@ -50,7 +50,7 @@ endif()
 # as we need the libraries as well.
 set(CMAKE_PREFIX_PATH_SAVED "${CMAKE_PREFIX_PATH}")
 unset(CMAKE_PREFIX_PATH)
-find_package(Python 3.10 COMPONENTS Interpreter REQUIRED)
+find_package(Python 3.9 COMPONENTS Interpreter REQUIRED)
 set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH_SAVED}")
 
 # Add the magic targets `check-*`
diff --git a/contrib/aur/lotus-qt/PKGBUILD b/contrib/aur/lotus-qt/PKGBUILD
index c079a2f4d1..ab7bb30eab 100644
--- a/contrib/aur/lotus-qt/PKGBUILD
+++ b/contrib/aur/lotus-qt/PKGBUILD
@@ -1,7 +1,7 @@
 # Maintainer: Josh Ellithorpe <quest@mac.com>
 
 pkgname=lotus-qt
-pkgver=0.23.1
+pkgver=1.0.0
 pkgrel=0
 pkgdesc="Lotus with lotusd, lotus-cli, lotus-tx, lotus-seeder and lotus-qt"
 arch=('i686' 'x86_64')
diff --git a/contrib/aur/lotus/PKGBUILD b/contrib/aur/lotus/PKGBUILD
index 03a423223e..d53f35e622 100644
--- a/contrib/aur/lotus/PKGBUILD
+++ b/contrib/aur/lotus/PKGBUILD
@@ -1,12 +1,12 @@
 # Maintainer: Josh Ellithorpe <quest@mac.com>
 
 pkgname=lotusd
-pkgver=0.34.1
+pkgver=1.0.0
 pkgrel=0
 pkgdesc="Lotus with lotusd, lotus-tx, lotus-seeder and lotus-cli"
 arch=('i686' 'x86_64')
 url="https://givelotus.org"
-depends=('boost-libs' 'libevent' 'openssl' 'zeromq' 'jemalloc')
+depends=('boost-libs' 'libevent' 'openssl' 'zeromq' 'miniupnpc' 'libnatpmp' 'jemalloc')
 makedepends=('cmake' 'ninja' 'boost' 'python' 'help2man')
 license=('MIT')
 source=(https://github.com/LogosFoundation/lotusd/archive/v$pkgver.tar.gz
